api/downloads/download-file.js

```diff
@@ -160,10 +160,10 @@ DownloadFile.prototype._onDownloadFailure = function (err, aborted) {
     if (this._errors <= this._options.maxDownloadRetry) {
       this._retryDownload();
     } else {
-      this.emit("error", err);
+      this.emit("error", {message: err});
```

Comment:
can you explain what it does and why is this change necessary ?

Reply:
Hi 
That is because onError callback is declared as : 
Download.prototype._onError = function (data) {
  const self = this;
  this.status = STATUSES.ERROR;
  data = data || {};
  const message = data.message || "";

  self._removeEvents();
  self._updateStats();
  self.events.emit("error", self, message);
};

Message is important to display or log the error.
But maybe it can be treated in another PR